stdlib/public/SDK/Foundation/TypePreservingNSNumber.mm

```diff
@@ -0,0 +1,253 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#import <Foundation/Foundation.h>
+#import <CoreGraphics/CoreGraphics.h>
+
+#include <swift/Runtime/Debug.h>
+
+#if __LP64__
+typedef int64_t SwiftIntTy;
+typedef uint64_t SwiftUIntTy;
+#else
+typedef int32_t SwiftIntTy;
+typedef uint32_t SwiftUIntTy;
+#endif
+
+@interface _SwiftTypePreservingNSNumber : NSNumber
+@end
+
+enum _SwiftTypePreservingNSNumberTag {
+  SwiftInt = 1,
+  SwiftUInt = 2,
+  SwiftFloat = 3,
+  SwiftDouble = 4,
+  SwiftCGFloat = 5,
+  SwiftBool = 6
+};
+
+@implementation _SwiftTypePreservingNSNumber {
+  @public _SwiftTypePreservingNSNumberTag tag;
+  @public char storage[8];
+}
+
+- (id)init {
```

Comment:
I'm calling this initializer from functions below, is there another way to create an instance?